var/spack/repos/builtin/packages/amdscalapack/package.py

```diff
@@ -47,8 +62,13 @@ def cmake_args(self):
             args.extend(['-DUSE_DOTC_WRAPPER:BOOL=%s' % (
                         'ON' if spec.satisfies('%aocc ^amdblis') else 'OFF')])
 
+        if "+ilp64" in spec:
+            args.extend(['-DENABLE_ILP64=ON'])
+
+        if "@:3.0" in spec:
+            args.extend(['-DUSE_F2C=ON'])
```

Comment:
You could use the cmake helpers (https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#adding-flags-to-cmake):

```suggestion
        args.extend([self.define_from_variant('ENABLE_ILP64', 'ilp64')])  # -DENABLE_ILP64:BOOL=ON
        args.extend([self.define('USE_F2C', spec.satisfies('@:3.0'))])  # -DUSE_F2C:BOOL=ON
```

Reply:
Incorporated review comments